soh/soh/Enhancements/debugconsole.cpp

```diff
@@ -1289,16 +1290,58 @@ static bool GenerateRandoHandler(std::shared_ptr<LUS::Console> Console, const st
     return 1;
 }
 
+static std::map<std::string, CosmeticGroup> cosmetic_groups = {
```

Comment:
If you switch this to a `std::array<std::pair<...>, N>` then you can turn this into a `static constexpr` and set the size to be `_MAX`. This will force a compile error in the future if anyone was to change the groups without changing these arrays.

Then you'd just need to tweak your for loops below to account for `std::pair`.

Reply:
I have no idea how to make a constexpr out of an array like that. How would I do that?

Reply:
ah looks like you cant use `std::string` with `constexpr`

So either do it without if we need `std::string`
```c++
static std::array<std::pair<const std::string, CosmeticGroup>, COSMETIC_GROUP_MAX> cosmetic_group = { {
    { "Link", COSMETIC_GROUP_LINK },
    { "Swords", COSMETIC_GROUP_SWORDS },
    ...
} };
```

Or switch to `const char*` to allow `constexpr`

```c++
static constexpr std::array<std::pair<const char*, CosmeticGroup>, COSMETIC_GROUP_MAX> cosmetic_group = { {
    { "Link", COSMETIC_GROUP_LINK },
    { "Swords", COSMETIC_GROUP_SWORDS },
    ...
} };
```